### DISCUSSION TITLE
build a world with radom obstacles

### DISCUSSION THREAD

**Discussion by ZiqiTIAN** [cyberbotics/webots#3715] [General]
Hello, Is there any methods to build a .wbt file with random obstacles? Now I could only open the simulation world which I had already designed, but what if trying to build a simulation whose obstacles are randomly located so every time we start the simulation, it is not as same as previous ones.

**Comment by stefaniapedrazzi** [comment]
Hi,
you can use a Supervisor controller to randomly place the obstacles at the very beginning of the simulation.
You can find an example in the `projects/samples/robotbenchmark/obstacle_avoidance/worlds/obstacle_avoidance.wbt` simulation, and in particular in this controller:
https://github.com/cyberbotics/webots/blob/master/projects/samples/robotbenchmark/obstacle_avoidance/controllers/supervisor/supervisor.py
